Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/ManipulationHandler.cs

```diff
@@ -756,6 +755,13 @@ private void ReleaseRigidBody()
             }
         }
 
+        private IMixedRealityPointer GetFirstPointer()
+        {
+            // We may be able to do this without allocating memory.
+            // Moving to a method for later investigation.
+            return pointerIdToPointerMap.Values.First();
```

Comment:
One minor improvement on this would be to cache this value anytime the pointer map is updated - so on pointer down/up, it will call .First() just once (which would then be reused for the duration of the move "updated" path).

More generally this would also lead to actual determinism in the algorithm as well, because right now it's only working thanks to luck - dictionary value enumeration is never advertised as being deterministic.

Reply:
I think that would be ideal. I'm going to leave that for @julenka to work out since she's most familiar with the state machine.